server/fleet/app.go

```diff
@@ -391,21 +391,35 @@ type Logging struct {
 }
 
 type UpdateIntervalConfig struct {
-	OSQueryDetail time.Duration `json:"osquery_detail"`
-	OSQueryPolicy time.Duration `json:"osquery_policy"`
+	OSQueryDetail Duration `json:"osquery_detail"`
+	OSQueryPolicy Duration `json:"osquery_policy"`
+}
+
+func (uic *UpdateIntervalConfig) Apply(osc config.OsqueryConfig) {
+	uic.OSQueryDetail = Duration{osc.DetailUpdateInterval}
+	uic.OSQueryPolicy = Duration{osc.PolicyUpdateInterval}
 }
 
 // VulnerabilitiesConfig contains the vulnerabilities configuration of the
 // fleet instance (as configured for the cli, either via flags, env vars or the
 // config file), not to be confused with VulnerabilitySettings which is the
 // configuration in AppConfig.
 type VulnerabilitiesConfig struct {
-	DatabasesPath         string        `json:"databases_path"`
-	Periodicity           time.Duration `json:"periodicity"`
-	CPEDatabaseURL        string        `json:"cpe_database_url"`
-	CVEFeedPrefixURL      string        `json:"cve_feed_prefix_url"`
-	CurrentInstanceChecks string        `json:"current_instance_checks"`
-	DisableDataSync       bool          `json:"disable_data_sync"`
+	DatabasesPath         string   `json:"databases_path"`
+	Periodicity           Duration `json:"periodicity"`
```

Comment:
Ah, didn't even remember we had this `Duration` type :)

So one thing we need is to make sure is that we parse ints here https://github.com/fleetdm/fleet/blob/ddc1d7b0aad223d9246c08574b42bdb3ff45a33c/server/fleet/app.go#L151 to allow for the configs with the old way to still work.

But before you do that, I suggest you start a conversation, maybe in slack is fine, mentioning this PR. The alternative is to "prettify" it at the fleetctl level, and leave the API the way it was before.

Reply:
Thanks Tomas, yeah I think it would be best to keep those feels as numbers and apply the proper formatting at the presentation layer. I'll update this.